File: 1138-grumpy-bookstore-owner/1138-grumpy-bookstore-owner.cpp
class Solution {
public:
    int maxSatisfied(vector<int>& customers, vector<int>& grumpy, int minutes) {
        int n = customers.size();
        int happyCustomers = 0;

        for(int i = 0; i<n; i++) {
            if(grumpy[i] == 0) {
                happyCustomers += customers[i];
                customers[i] = 0;
            }
        }

        int maxSum = 0, sum = 0;;
        for(int i = 0; i<n; i++) {
            sum += customers[i];
            if(i >= minutes) {
                sum -= customers[i-minutes];
            }
            maxSum = max(maxSum, sum);
        }

        return happyCustomers + maxSum;
    }
};

File: 1138-grumpy-bookstore-owner/1138-grumpy-bookstore-owner.py
class Solution(object):
    def maxSatisfied(self, customers, grumpy, minutes):
        n = len(customers)
        happy_customers = 0
        for i in range(n):
            if grumpy[i] == 0:
                happy_customers += customers[i]
                customers[i] = 0
        
        max_sum , _sum = 0, 0
        for i in range(n):
            _sum += customers[i]
            if i >= minutes:
                _sum -= customers[i-minutes]
            max_sum = max(max_sum, _sum)
        
        return max_sum + happy_customers
        